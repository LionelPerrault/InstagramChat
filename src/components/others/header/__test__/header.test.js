import React from 'react'
import Header from '../header'
import { create } from 'react-test-renderer'
import { mount } from 'enzyme'

describe('Header component', () => {
  const comp = <Header />

  it('should match snapshot', () => {
    const tree = create(comp).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should show options when clicked', () => {
    const wrapper = mount(comp)

    expect(wrapper.find('HeaderOptions').exists()).toBeFalse()
    wrapper.find('.show_more').simulate('click')
    expect(wrapper.find('HeaderOptions').exists()).toBeTrue()
  })
})
