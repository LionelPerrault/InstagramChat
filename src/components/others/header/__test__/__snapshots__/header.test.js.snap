// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match snapshot 1`] = `
<div
  className="header"
>
  <div
    className="logo"
  >
    <Link
      to="/"
    >
      <img
        alt="Instagram"
        src="/images/instagram.jpg"
      />
    </Link>
  </div>
  <div>
    <div
      className="search_box"
    >
      <input
        autoComplete="false"
        autoFocus={true}
        className="search"
        disabled={false}
        maxLength="255"
        onChange={[Function]}
        placeholder="Search Instagram"
        spellCheck="false"
        type="text"
        value=""
      />
      <span
        className="search_icon"
      >
        <i
          className="fas fa-search"
        />
      </span>
    </div>
  </div>
  <div
    className="header_right"
  >
    <NavLink
      activeClassName="ha_active"
      className="notification"
      to="/notifications"
    >
      <span
        className="notification_span nav_icon"
      >
        <i
          className="material-icons"
        >
          notifications_none
        </i>
      </span>
      <span
        className="links_span"
      >
        Notifications
      </span>
    </NavLink>
    <NavLink
      activeClassName="ha_active"
      className="sp"
      to="/profile/null"
    >
      <img
        alt="avatar"
        className="sp_img"
        src="/users/null/avatar.jpg"
      />
      <span
        className="sp_span"
      />
    </NavLink>
    <span
      className="show_more"
      onClick={[Function]}
    >
      <i
        className="material-icons"
      >
        expand_more
      </i>
    </span>
  </div>
</div>
`;
